code/edge/src/main/java/com/adobe/marketing/mobile/EdgeExtension.java

```diff
@@ -573,33 +387,14 @@ Collection<Event> getCachedEvents() {
 	 * @return the Configuration shared state or null if it is pending
 	 */
 	private Map<String, Object> getConfig(final Event event) {
-		Map<String, Object> configSharedState = getApi()
-			.getSharedEventState(
-				EdgeConstants.SharedState.CONFIGURATION,
-				event,
-				new ExtensionErrorCallback<ExtensionError>() {
-					@Override
-					public void error(ExtensionError extensionError) {
-						MobileCore.log(
-							LoggingMode.WARNING,
-							LOG_TAG,
-							"EdgeExtension - Failed to retrieve Configuration shared state with error " +
-							extensionError.getErrorName()
-						);
-					}
-				}
-			);
-
-		// configuration is required for edge.configId, so if shared state is null (pending) stop processing events
-		if (configSharedState == null) {
-			MobileCore.log(
-				LoggingMode.DEBUG,
-				LOG_TAG,
-				"EdgeExtension - Configuration is pending, couldn't process event at this time, waiting..."
-			);
+		SharedStateResult sharedStateResult = getApi()
+			.getSharedState(EdgeConstants.SharedState.CONFIGURATION, event, false, SharedStateResolution.ANY);
+		if (sharedStateResult == null || sharedStateResult.getStatus() != SharedStateStatus.SET) {
+			Log.debug(LOG_TAG, LOG_SOURCE, "Configuration is pending, couldn't process event at this time, waiting...");
```

Comment:
this should not be the case as we check in ready for event, but this message is now confusing as there is no more waiting at this time since the event was already processed and there is no retry after that. I suggest we raise this to a warning mentioning this is unexpected and remove the waiting part

Reply:
I reorganized this a little to make it cleaner. I moved the warning messages to the (soon to be renamed) processEdgeEvent method (same as in iOS) and then I'm using these getter in readyForEvent as well so the code to retrieve the shared states is the same both when it is checked (readyForEvent) and when received (processEdgeEvent).